tests/lax_numpy_reducers_test.py

```diff
@@ -770,5 +770,72 @@ def test_f16_mean(self, dtype):
     self.assertAllClose(expected, actual, atol=0)
 
 
+
+  def _is_canonical(dtype):
+    _dtype = dtypes.dtype(dtype)
+    return _dtype == dtypes.canonicalize_dtype(_dtype)
+
+  @jtu.sample_product(
+    [dict(shape=shape, axis=axis)
+      for shape in all_shapes
+      for axis in list(
+        range(-len(shape), len(shape))
+      ) + ([None] if len(shape) == 1 else [])],
+    dtype=filter(_is_canonical, all_dtypes),
+    out_dtype=filter(_is_canonical, all_dtypes),
+    include_initial=[False, True],
+  )
+  @jtu.ignore_warning(category=NumpyComplexWarning)
+  @jax.numpy_dtype_promotion('standard')  # This test explicitly exercises mixed type promotion
+  def testCumulativeSum(self, shape, axis, dtype, out_dtype, include_initial):
+    rng = jtu.rand_some_zero(self.rng())
+
+    def np_mock_fun(x, axis=None, dtype=None, include_initial=False):
+      kind = x.dtype.kind
+      if (dtype is None and kind in {'i', 'u'}
+          and x.dtype.itemsize*8 < int(config.default_dtype_bits.value)):
+        dtype = dtypes.canonicalize_dtype(dtypes._default_types[kind])
+      axis = axis or 0
+      x = x.astype(dtype=dtype or x.dtype)
+      out = jnp.cumsum(x, axis=axis)
+      if include_initial:
+        zeros_shape = list(x.shape)
+        zeros_shape[axis] = 1
+        out = jnp.concat([jnp.zeros(zeros_shape, dtype=out.dtype), out], axis=axis)
+      return out
+
+
+    # We currently "cheat" to ensure we have JAX arrays, not NumPy arrays as
+    # input because we rely on JAX-specific casting behavior
+    args_maker = lambda: [jnp.array(rng(shape, dtype))]
+    np_op = getattr(np, "cumulative_sum", np_mock_fun)
+    kwargs = dict(axis=axis, dtype=out_dtype, include_initial=include_initial)
+    np_fun = lambda x: np_op(x, **kwargs)
+    jnp_fun = lambda x: jnp.cumulative_sum(x, **kwargs)
+    self._CheckAgainstNumpy(np_fun, jnp_fun, args_maker)
+    self._CompileAndCheck(jnp_fun, args_maker)
+
+    kwargs = dict(axis=axis, include_initial=include_initial)
```

Comment:
I'd avoid this kind of action at a distance (changing a global value in order to change the behavior of a previously-defined function). Maybe just include `out_dtype = None` in the parameterization, so you only have one pair of functions to check per test run?

Reply:
> I'd avoid this kind of action at a distance (changing a global value in order to change the behavior of a previously-defined function)

I have updated the approach to avoid the indirect parameterization.

> Maybe just include `out_dtype = None` in the parameterization, so you only have one pair of functions to check per test run?

Because there are some important nuances with `dtype=None` based on various input dtypes, I think it is worth explicitly testing `dtype=None` _in each test instance_, otherwise w/ only 10 generated samples per test, even across CI there's a chance that we may have insufficient coverage. What do you think?

Reply:
We run 90 samples per test internally, so I think it's probably OK.